test/e2e/aws_test.go

```diff
@@ -135,10 +172,35 @@ func (a *awsTestHelper) deleteHostedZone(zoneID string) error {
 	}
 
 	zoneInput := route53.DeleteHostedZoneInput{
-		Id: &zoneID,
+		Id: &hostedZoneID,
 	}
 	if _, err := a.r53Client.DeleteHostedZone(&zoneInput); err != nil {
 		return err
 	}
 	return nil
 }
+
+func prepareConfigurations(isOpenShiftCI bool) (awsAccessKeyID, awsSecretAccessKey string, err error) {
+	if isOpenShiftCI {
+		awsAccessKeyID, awsSecretAccessKey, err = rootCredentials()
+		if err != nil {
+			return "", "", fmt.Errorf("failed to get AWS credentials: %w", err)
+		}
+	} else {
+		awsAccessKeyID = mustGetEnv("AWS_ACCESS_KEY_ID")
+		awsSecretAccessKey = mustGetEnv("AWS_SECRET_ACCESS_KEY")
+	}
+	return
+}
+
+func rootCredentials() (string, string, error) {
+	secret := &corev1.Secret{}
+	secretName := types.NamespacedName{
+		Name:      "aws-creds",
+		Namespace: "kube-system",
+	}
+	if err := kubeClient.Get(context.TODO(), secretName, secret); err != nil {
+		return "", "", fmt.Errorf("failed to get credentials secret %s: %w", secretName.Name, err)
+	}
```

Comment:
This part is repeated in each provider, let's move the retrieval of the secret to `utils.go`.

Reply:
Done